src/Common/src/TypeSystem/Common/TypeSystemContext.cs

```diff
@@ -120,9 +120,11 @@ protected override bool CompareKeyToValue(ArrayTypeKey key, ArrayType value)
                     if (key._elementType != value.ElementType)
                         return false;
 
-                    if ((key._rank == -1) && value.IsSzArray)
-                        return true;
-
+                    if (value.IsSzArray)
+                    {
```

Comment:
Nit... if statements with a single statement in their conditional block should not have braces